CMakeLists.txt

```diff
@@ -301,27 +288,12 @@ if(ENABLE_SHARED)
   # More information on RPATH in CMake:
   # https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/RPATH-handling
   set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR};${PROJECT_SOURCE_DIR}/deps/install/lib")
-else()
-  # When building statically, we *only* want static archives/libraries
-  if (WIN32)
-      set(CMAKE_FIND_LIBRARY_SUFFIXES .lib .a)
-  else()
-      set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
-  endif()
```

Comment:
We don't need this anymore?

Reply:
This is a bit more complex now, as "some libraries" (gmp) need to modify it: as the static build of GMP is not build with `-fPIC` it is not suitable for dynamic linking and we thus need both `libgmp.a` and `libgmp.so`. It has been replaced by a set of macros now, that some `FindX.cmake` scripts may use.

Reply:
What do we do if the system provides a lib.a that is not compiled with `-fPIC`?

Reply:
This is now the responsibility of the `FindX` script. Most libraries can simply export the target `X` that holds the static library and works for both static and dynamic builds (because it's build with `-fPIC`). If it is not, the `FindX` script is expected to create targets `X-shared` and `X-static`, as `FindGMP` does. Of course our usage of these targets (when linking them) needs to be aware of what type of find-script it is...